### PR TITLE
[DOC] Improve warnings for BIDS folders

### LINKED ISSUE #4252: improve warning for data folder in `first_level_from_bids`
warning for no subject found are not super helpful if they do not say in which folder the subject was not found

```python
from pathlib import Path

from nilearn.glm.first_level import first_level_from_bids


data_dir = Path(__file__).parent.parent / "outputs" / "study-ds000208"
task_label = "rest"
space_label = "MNI152NLin2009cAsym"

# data is in
# derivatives_folder = "derivatives/fmriprep-21.0.2"
# and not the default "derivatives"

(
    models,
    models_run_imgs,
    models_events,
    models_confounds,
) = first_level_from_bids(
    data_dir,
    task_label,
    space_label,
    sub_labels=['01'],
    verbose=1,
    n_jobs=2,
)

```

Getting 

```
/home/remi/github/nilearn/nilearn/nilearn/interfaces/bids/query.py:141: UserWarning: No bold.json found in BIDS folder in /home/remi/github/origami/cohort_creator/demo/outputs/study-ds000208/derivatives.
  warn(f"No bold.json found in BIDS folder{msg_suffix}.")
/home/remi/github/nilearn/nilearn/nilearn/interfaces/bids/query.py:141: UserWarning: No bold.json found in BIDS folder in /home/remi/github/origami/cohort_creator/demo/outputs/study-ds000208.
  warn(f"No bold.json found in BIDS folder{msg_suffix}.")
/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/first_level.py:1335: UserWarning: 't_r' not provided and cannot be inferred from BIDS metadata.
It will need to be set manually in the list of models, otherwise their fit will throw an exception.
  warn(
/home/remi/github/nilearn/nilearn/nilearn/interfaces/bids/query.py:96: UserWarning: No bold.json found in BIDS folder in /home/remi/github/origami/cohort_creator/demo/outputs/study-ds000208/derivatives.
  warn(f"No bold.json found in BIDS folder{msg_suffix}.")
/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/first_level.py:1496: UserWarning: Subject label 01 is not present in the dataset and cannot be processed.
  warn(
```

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4252

Changes proposed in this pull request:

- indicate folder where subjects were not found
- increase stack level of several warnings
- add error when no subject was found
- use absolute path

### CODE REVIEW

File: nilearn/glm/first_level/first_level.py [hunk 1]
```diff
@@ -1494,8 +1496,10 @@ def _list_valid_subjects(derivatives_path, sub_labels):
             sub_labels_exist.append(sub_label_)
         else:
             warn(
-                f"Subject label {sub_label_} is not present in the"
-                " dataset and cannot be processed."
+                f"\nSubject label {sub_label_} is not present "
+                "in the following dataset and cannot be processed:\n"
+                f" {derivatives_path}",
```
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:1501]: "main" change

**Comment by Remi-Gau** [comment]
@michellewang

let me know if this makes sense

**Comment by Remi-Gau** [comment]
example output of a failed run

```
/home/remi/github/origami/cohort_creator/demo/code/run_glm.py:16: UserWarning: 
Subject label aa is not present in the following dataset and cannot be processed:
 /home/remi/github/origami/cohort_creator/demo/outputs/study-ds000208/derivatives/fmriprep-21.0.2
  ) = first_level_from_bids(
Traceback (most recent call last):
  File "/home/remi/github/origami/cohort_creator/demo/code/run_glm.py", line 16, in <module>
    ) = first_level_from_bids(
        ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/first_level.py", line 1403, in first_level_from_bids
    raise RuntimeError(f"\nNo subject found in:\n {derivatives_path}")
RuntimeError: 
No subject found in:
 /home/remi/github/origami/cohort_creator/demo/outputs/study-ds000208/derivatives/fmriprep-21.0.2
```

**Review by bthirion — APPROVED** [review]
Did not sport any issue. Thx !

**Comment by michellewang** [comment]
> @michellewang
let me know if this makes sense

Realized that I forgot to get back to you 😅 but yes the error message looks good to me!

**Pull request merged**